src/identity/components/IdentityLayout.tsx

```diff
@@ -35,7 +35,9 @@ export function IdentityLayout({
     return {
       avatar: childrenArray.find(findComponent(Avatar)),
       name: childrenArray.find(findComponent(Name)),
-      address: childrenArray.find(findComponent(Address)),
+      address: childrenArray.find(findComponent(Address))
+        ? cloneElement(childrenArray.find(findComponent(Address))!, { onClick })
+        : undefined,
```

Comment:
you could move this childrenArray.find up above your return and not have to loop twice